### PR TITLE
CDAP-3639: Moving ArtifactDescriptor to app-fabric

### PR DESCRIPTION
**Pull request by gokulavasan**
PluginSelector should have access only to ArtifactIds. Moving ArtifactDescriptor to cdap-app-fabric.

JIRA : https://issues.cask.co/browse/CDAP-3639

Build : http://builds.cask.co/browse/CDAP-DUT2817

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java [hunk 1]
```diff
@@ -235,7 +237,23 @@ public ArtifactDetail getArtifact(Id.Artifact artifactId) throws IOException, Ar
...
+    SortedMap<ArtifactId, PluginClass> artifactIds = Maps.newTreeMap();
+    for (Map.Entry<ArtifactDescriptor, PluginClass> pluginClassEntry : pluginClasses.entrySet()) {
+      artifactIds.put(pluginClassEntry.getKey().getArtifactId(), pluginClassEntry.getValue());
+    }
+    Map.Entry<ArtifactId, PluginClass> chosenArtifact = selector.select(artifactIds);
+    if (chosenArtifact == null) {
+      throw new PluginNotExistsException(artifactId, pluginType, pluginName);
+    }
+
+    for (Map.Entry<ArtifactDescriptor, PluginClass> pluginClassEntry : pluginClasses.entrySet()) {
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java]: we can't just call get on the map? ArtifactDescriptor doesn't use location in its equality and comparison implementations right?

Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java]: er nevermind, was confused, keys are different

Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java]: i guess we could do a get with a null location... not sure how we could simplify this

**Comment by albertshau** [comment]
would be nice to avoid all the copying, though not sure how we would do that. lgtm

**Comment by gokulavasan** [comment]
Build is :green_apple:. Merging.

**Pull request merged**